pkg/cmd/workspace/code.go

```diff
@@ -204,3 +204,78 @@ func init() {
 	ideListStr := strings.Join(ids, ", ")
 	CodeCmd.Flags().StringVarP(&ideFlag, "ide", "i", "", fmt.Sprintf("Specify the IDE (%s)", ideListStr))
 }
+
+func StartIDE(ideId string, activeProfile config.Profile, workspaceId string, workspaceName string, projectName string, providerMetadata string, verbose bool) error {
+	if ideId == "" {
+		c, err := config.GetConfig()
+		if err != nil {
+			return err
+		}
+		ideId = c.DefaultIdeId
+	}
+
+	if projectName == "" {
+		selectedProject, err := selectWorkspaceProject(workspaceName, &activeProfile)
+		if err != nil {
+			return err
+		}
+		if selectedProject == nil {
+			return errors.New("project not found")
+		}
+		projectName = selectedProject.Name
+	}
+
+	if providerMetadata == "" || workspaceId == "" {
+		workspace, err := apiclient_util.GetWorkspace(workspaceName)
+		if err != nil {
+			if strings.Contains(err.Error(), workspaces.ErrWorkspaceNotFound.Error()) {
+				log.Debug(err)
+				return errors.New("workspace not found. You can see all workspace names by running the command `daytona list`")
+			}
+			return err
+		}
+
+		workspaceName = workspace.Name
+		workspaceId = workspace.Id
+
+		if ideId != "ssh" {
+			providerMetadata, err = GetProjectProviderMetadata(workspace, projectName)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	if verbose {
+		ideList := config.GetIdeList()
+		ideName := ""
+		for _, ide := range ideList {
+			if ide.Id == ideId {
+				ideName = ide.Name
+				break
+			}
+		}
+		views.RenderInfoMessage(fmt.Sprintf("Opening the project '%s' from workspace '%s' in %s", projectName, workspaceName, ideName))
+	}
+
+	err := openIDE(ideId, activeProfile, workspaceId, projectName, providerMetadata)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func GetProjectProviderMetadata(workspace *apiclient.WorkspaceDTO, projectName string) (string, error) {
```

Comment:
This function can be moved to some util package and reused in the `code.go` and `create.go` files.

Reply:
done

Reply:
You didn't add the util function to `code.go` (in the `Run` method)

Reply:
I have done the Run method changes in this PR: https://github.com/daytonaio/daytona/pull/991/files#diff-9a18befd36b9809df9eeece74c32458be188ab67d9c0bfca81a69294a4b23499R111-R114